src/contour/Config.h

```diff
@@ -144,138 +150,332 @@ struct SimpleColorConfig
+        vtbackend::Modifier mouseBlockSelectionModifier = vtbackend::Modifier::Control;
+
+        // input mapping
+        InputMappings inputMappings;
+
+        bool spawnNewProcess = false;
+        std::shared_ptr<logstore::sink> loggingSink;
+
+        bool sixelScrolling = true;
+        vtbackend::ImageSize maxImageSize = {}; // default to runtime system screen size.
+        unsigned maxImageColorRegisters = 4096;
+
+        std::set<std::string> experimentalFeatures;
+    };
+} // namespace old
+
+template <typename T>
+struct ConfigEntry
 {
-    std::filesystem::path backingFilePath;
+    using val_type = T;
 
-    bool live = false;
+    constexpr ConfigEntry(std::string d): _val {}, _documentation { std::move(d) } {}
 
-    /// Qt platform plugin to be loaded.
-    /// This is equivalent to QT_QPA_PLATFORM.
-    std::string platformPlugin;
+    constexpr ConfigEntry(T&& in, std::string d):
+        _val { std::forward<T>(in) }, _documentation { std::move(d) }
+    {
+    }
 
-    RenderingBackend renderingBackend = RenderingBackend::Default;
+    template <typename F>
+        requires std::is_constructible_v<T, F>
+    constexpr ConfigEntry(F in, std::string d): _val { static_cast<T>(in) }, _documentation { std::move(d) }
+    {
+    }
 
-    /// Enables/disables support for direct mapped texture atlas tiles (e.g. glyphs).
-    bool textureAtlasDirectMapping = true;
+    [[nodiscard]] constexpr T const& get() const { return _val; }
+    [[nodiscard]] constexpr T& get() { return _val; }
 
-    // Number of hashtable slots to map to the texture tiles.
-    // Larger values may increase performance, but too large may also decrease.
-    // This value is rounted up to a value equal to the power of two.
-    //
-    // Default: 4096
-    crispy::strong_hashtable_size textureAtlasHashtableSlots = crispy::strong_hashtable_size { 4096 };
+    [[nodiscard]] constexpr std::string makeDoc() const
+    {
+        return fmt::format(fmt::runtime(_documentation), _val);
+    }
 
-    /// Number of tiles that must fit at lest into the texture atlas,
-    /// excluding US-ASCII glyphs, cursor shapes and decorations.
-    ///
-    /// Value must be at least as large as grid cells available in the current view.
-    /// This value is automatically adjusted if too small.
-    crispy::lru_capacity textureAtlasTileCount = crispy::lru_capacity { 4000 };
+    constexpr ConfigEntry(ConfigEntry const&) = default;
+    constexpr ConfigEntry& operator=(ConfigEntry const&) = default;
+    constexpr ConfigEntry(ConfigEntry&&) noexcept = default;
+    constexpr ConfigEntry& operator=(ConfigEntry&&) noexcept = default;
+    ~ConfigEntry() = default;
 
-    // Configures the size of the PTY read buffer.
-    // Changing this value may result in better or worse throughput performance.
-    //
-    // This value must be integer-devisable by 16.
-    size_t ptyReadBufferSize = 16384;
+  private:
+    std::string _documentation;
+    T _val;
```

Comment:
```suggestion
    T _value;
```
let's please avoid abbreviations, if possible :)